docs/_build/html/_sources/write_your_tiltfile.md.txt

```diff
@@ -0,0 +1,79 @@
+# Write Your Tiltfile
+
+A Tiltfile can be broken down in to three things: Kubernetes YAML, Docker images and resources. In this doc we'll step through all three concepts so you can learn to write your own Tiltfiles with gusto!
+
+To start here's an illustrative example that uses most of the features of Tilt.
+
+```python
+## Part 1: Kubernetes YAML
+k8s_yaml("k8s.yaml")               # one yaml file
+k8s_yaml(['foo.yaml', 'bar.yaml']) # list of yaml files
+k8s_yaml(kustomize('config_dir'))  # run kustomize to generate yaml
+k8s_yaml(local('gen_k8s_yaml.py')) # run a custom command to general yaml
+
+## Part 2: Images
+docker_build("companyname/frontend", "frontend") # docker build <path>
+docker_build("companyname/backend", "backend", dockerfile="backend/Dockerfile.dev") # docker build with specific Dockerfile
+docker_build("companyname/graphql", "graphql", args={"target": "local"} # docker build with build args
+
+## Part 3: Resources
+k8s_resource("backend", img="companyname/backend/server") # give a resource a name that's different than the base name of the image
+k8s_resource("frontend", port_forwards=9000) # connect to a specific local port
+```
+
+Let's dig in to each of these sections in detail.
+
+## Part 1: Kubernetes YAML
+Start  by telling Tilt about your Kubernetes YAML:
+
+```python
+k8s_yaml("k8s.yaml")               # one yaml file
+```
+
+Now Tilt will parse the YAML and deploy. A `k8s_yaml` function call is all you need to have a functioning Tiltfile.
+
+Sometimes you organize your YAML in to multiple files or generate it via a script. Tilt supports those cases too:
+
+```python
+k8s_yaml(['foo.yaml', 'bar.yaml']) # list of yaml files
+k8s_yaml(kustomize('config_dir'))  # run kustomize to generate yaml
+k8s_yaml(local('gen_k8s_yaml.py')) # run a custom command to general yaml
+```
+
+## Part 2: Images
+Next tell Tilt about all your Dockerfiles. Tilt can deploy any YAML but it watches resources by image name. You tell Tilt to build a Dockerfile like so:
```

Comment:
This is really confusing. I'm kind of struggling to offer a suggestion because I'm not even sure what concept it's trying to describe.

Maybe something like this?

Tilt will deploy whatever is defined in the yaml you give it, but most of its functionality requires knowing about images.

Reply:
is this referring to tilt resources before we defined them later? maybe a hyperlink or "(see below)"?